### DISCUSSION TITLE
Saving black and white images of molecule

### DISCUSSION THREAD

**Discussion by kot239** [rdkit/rdkit#4602] [Q&A] [answered]
Hi everyone!
I am trying to do my project and I have one problem now. I want to save molecule as black and white image with rdkit.Chem.Draw.MolToFile function, but I can't find any information about how to do it. Also I've found this example https://www.rdkit.org/docs/Cookbook.html#black-and-white-molecules , but it didn't help me :(

**Comment by DaShenZi721** [comment]
Hi @kot239 ,

I think you could save molecule as color image first, then convert it to black and white image with PIL.

```
from PIL import Image
from rdkit import Chem
from rdkit.Chem import Draw

mol = Chem.MolFromSmiles('OC(=O)c1[te]ccc1')
canvas = Draw.rdMolDraw2D.MolDraw2DCairo(*(300, 300))
Draw.rdMolDraw2D.PrepareAndDrawMolecule(canvas, mol)
canvas.WriteDrawingText('foo.png')
img = Image.open('foo.png')
img = img.convert('1')
img.save('foo.png')
```
![foo](https://user-images.githubusercontent.com/39253334/137135598-3af85d61-592d-4508-9784-3dbc62012eea.png)

**Comment by dehaenw** [comment, marked as answer]
Hello @kot239,
because PIL gives an image that has some artifacts, it's preferable to set a custom palette inside RDKit where everything is black before rendering, like this: (just one line added to @DaShenZi721 example)
```python
from rdkit import Chem
from rdkit.Chem import Draw

mol = Chem.MolFromSmiles('OC(=O)c1[te]ccc1')
canvas = Draw.rdMolDraw2D.MolDraw2DCairo(*(300, 300))
canvas.drawOptions().setAtomPalette({-1:(0,0,0)})
Draw.rdMolDraw2D.PrepareAndDrawMolecule(canvas, mol)
canvas.WriteDrawingText('foo.png')
```
![image](https://user-images.githubusercontent.com/66372095/137157057-613850f0-a88f-404b-bd83-238b8600073c.png)

strangely enough, i cant get useBWAtomPalette() working either. but the above option works!

**Reply by greglandrum** [reply]
what goes wrong when you use `useBWAtomPalette()`?
This works fine for me:
```
mol = Chem.MolFromSmiles('OC(=O)c1[te]ccc1')
canvas = Draw.rdMolDraw2D.MolDraw2DCairo(*(300, 300))
canvas.drawOptions().useBWAtomPalette()
Draw.rdMolDraw2D.PrepareAndDrawMolecule(canvas, mol)
canvas.WriteDrawingText('/tmp/foo.png')
```

**Reply by dehaenw** [reply]
The above also works on my end. On closer inspection, the "issue" on my side turned out to be an error in my code, because i was changing the palette not on the level of the canvas but instead on the MolDrawOptions and then not applying the options to the canvas. I had used 
```python
mol = Chem.MolFromSmiles('OC(=O)c1[te]ccc1')
canvas = Draw.rdMolDraw2D.MolDraw2DCairo(*(300, 300))
Draw.MolDrawOptions().useBWAtomPalette()
Draw.rdMolDraw2D.PrepareAndDrawMolecule(canvas, mol)
canvas.WriteDrawingText('foo.png')
```
whereas if i wanted to set the palette in this way i should have used this:
```python
mol = Chem.MolFromSmiles('OC(=O)c1[te]ccc1')
canvas = Draw.rdMolDraw2D.MolDraw2DCairo(*(300, 300))
options = Draw.MolDrawOptions()
options.useBWAtomPalette()
canvas.SetDrawOptions(options)
Draw.rdMolDraw2D.PrepareAndDrawMolecule(canvas, mol)
canvas.WriteDrawingText('foo.png')
```

**Reply by greglandrum** [reply]
Ah yeah, that explains it.
Glad that you have something that works now.
